### PR TITLE
WIP: Upgrade Kotlin / KSP / Compose Multiplatform

### LINKED ISSUE #547: Upgrade Compose Multiplatform / Kotlin
Upgrade the following dependencies:
- Compose Multiplatform to `1.4.3`
- Kotlin to `1.9.0`
- KSP to `1.9.0-1.0.12`

These upgrades will bring the latest features and improvements to our project while ensuring compatibility with Kotlin 1.9.0 and the latest KSP version.

### PR DESCRIPTION
**Pull request by manuel-martos**
## Description

Fixes #547 

## Check list

- [ ] I have updated `CHANGELOG.md` if required.
- [ ] I have updated documentation if required.

### CODE REVIEW

File: utils/interop-ribs/lint-baseline.xml [hunk 1]
```diff
@@ -1,4 +1,15 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<issues format="6" by="lint 7.1.3" type="baseline" client="gradle" dependencies="false" name="AGP (7.1.3)" variant="all" version="7.1.3">
+<issues format="6" by="lint 7.4.1" type="baseline" client="gradle" dependencies="false" name="AGP (7.4.1)" variant="all" version="7.4.1">
+
+    <issue
+        id="NewApi"
```
Comment (KovalevAndrey) [inline utils/interop-ribs/lint-baseline.xml]: I had this issue before, it went away with agp update

Comment (manuel-martos) [reply utils/interop-ribs/lint-baseline.xml]: Which AGP version should remove this false positive?

Comment (KovalevAndrey) [reply utils/interop-ribs/lint-baseline.xml]: I think it wasn't a specific version but rather combination kotlin + agp

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -9,16 +9,16 @@ androidTargetSdk = "32"
 androidx-lifecycle = "2.6.1"
 androidx-navigation-compose = "2.5.1"
 coil = "2.2.1"
-composePlugin = "1.4.0"
+composePlugin = "1.5.0"
```
Comment (KovalevAndrey) [inline gradle/libs.versions.toml:12]: but we can't update it without updating google compose to 1.5? And compose update at the main project is currently blocked

**Comment by zsoltk** [comment]
Closing as done with even later versions in #662

**Pull request closed without merging**